core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -242,6 +256,29 @@ private[spark] class CoarseGrainedExecutorBackend(
 
     System.exit(code)
   }
+
+  private def decommissionSelf(): Boolean = {
+    logInfo("Decommissioning self w/sync")
+    try {
+      decommissioned = true
+      // Tell master we are are decommissioned so it stops trying to schedule us
+      if (driver.nonEmpty) {
+        driver.get.askSync[Boolean](DecommissionExecutor(executorId))
```

Comment:
Instead of decommission Executor, Can we have Entire node decommission
eg.  driver.get.askSync[Boolean](AddNodeToDecommission(hostname,  terminationTimeMs , SpotLoss))

Reply:
Same as previous comment, in standalone only sure, but in YARN/K8s we could see individual executors decommission.